File: pinyougou-sellergoods-interface/src/main/java/com/xt/pinyougou/service/ItemCatService.java
package com.xt.pinyougou.service;

import com.xt.pinyougou.entity.ItemCat;
import com.baomidou.mybatisplus.extension.service.IService;

import java.util.List;

/**
 * <p>
 * 商品类目 服务类
 * </p>
 *
 * @author xt
 * @since 2020-01-07
 */
public interface ItemCatService extends IService<ItemCat> {
    /**
     * 根据上级ID返回列表
     */
    List<ItemCat> findByParentId(Long parentId);

    boolean deleteBatch(List<Long> ids);
}

File: pinyougou-shop-web/src/main/resources/static/service/itemCatService.js
// 商品分类管理 服务
app.service('itemCatService', function ($http) {

    // 根据上级ID查询下级列表
    this.findByParentId = function (parentId) {
        return $http.get('/consumer/itemCat/findByParentId/' + parentId);
    };

    // 查询详情
    this.findOne = function (id) {
        return $http.get('/consumer/itemCat/' + id);
    };

    // 查询所有列表
    this.findItemCatList = function () {
        return $http.get('/consumer/itemCat/list');
    }
});